webservice/src/main/java/com/hortonworks/registries/webservice/RegistryApplication.java

```diff
@@ -90,6 +87,33 @@ public void run(RegistryConfiguration registryConfiguration, Environment environ
 
     }
 
+    private void initializeUGI(RegistryConfiguration conf) throws IOException {
+        LOG.debug("Initialization of User Group ininformation...");
+        if (UserGroupInformation.isSecurityEnabled()) {
+            LOG.debug("UGI.isSecurityEnabled() = true.");
+
+            AuthMethodConfiguration authMethodConf = conf.getAuthenticationMethod();
+            if(authMethodConf != null) {
+                String serverPrincipal = authMethodConf.getServerPrinciple();
+                String keyTab = authMethodConf.getServerPrincipleKeytab();
+
+                LOG.debug("UGI is trying to login with principle = " + serverPrincipal
+                        + ", keyTab = " + keyTab);
+
+                //Authenticate using keytab
+                UserGroupInformation.loginUserFromKeytab(serverPrincipal, keyTab);
```

Comment:
Why cannot we use KerberosLogin class from common-auth module?

Reply:
Usage of UserGroupInformation is required by the ranger internal implementation.